### PR TITLE
#3038 Include operation name ApolloNetworkException

### PR DESCRIPTION
**Pull request by semaphore3000**
Include operation name in ApolloNetworkException for better debugability.

### CODE REVIEW

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java [hunk 1]
```diff
@@ -97,16 +97,20 @@ public void interceptAsync(@NotNull final InterceptorRequest request, @NotNull f
           }
 
         } catch (IOException e) {
-          logger.e(e, "Failed to prepare http call for operation %s", request.operation.name().name());
-          callBack.onFailure(new ApolloNetworkException("Failed to prepare http call", e));
+          final String operationName = request.operation.name().name();
+          final String message = String.format("Failed to prepare http call for operation [%s]", operationName);
```
Comment (martinbonnin) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java:101]: Nitpick as we're using single quotes in other places
```suggestion
          final String message = String.format("Failed to prepare http call for operation '%s'", operationName);
```

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java [hunk 2]
```diff
@@ -97,16 +97,20 @@ public void interceptAsync(@NotNull final InterceptorRequest request, @NotNull f
...
           return;
         }
 
         httpCall.enqueue(new Callback() {
           @Override public void onFailure(@NotNull Call call, @NotNull IOException e) {
             if (disposed) return;
-            logger.e(e, "Failed to execute http call for operation %s", request.operation.name().name());
-            callBack.onFailure(new ApolloNetworkException("Failed to execute http call", e));
+            final String operationName = request.operation.name().name();
+            final String message = String.format("Failed to execute http call for operation [%s]", operationName)
```
Comment (martinbonnin) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java:111]: Same nitpick
```suggestion
            final String message = String.format("Failed to execute http call for operation '%s'", operationName)
```

**Review by martinbonnin — APPROVED** [review]
LGTM! 
For some reason it looks like this has a conflict with `main`. Do you mind rebasing ?

**Comment by semaphore3000** [comment]
Looks like this is some master / main confusion. Locally my main branch is still "master". I will need some time to figure it out.

**Comment by martinbonnin** [comment]
> Looks like this is some master / main confusion. Locally my main branch is still "master". I will need some time to figure it out.

Yep, we renamed to `main` some time ago. Let me know how that goes, if that's too much of a hassle, I can certainly forward port your changes. Just let me know.

**Comment by semaphore3000** [comment]
Yes, please port it. I can´t figure it out ATM, I only see the orphaned master and no main branch.

The changes should be 

```
      val operationName = request.operation.name().name()
      val message = "Failed to prepare http call for operation '$operationName'“
      logger.e(e, message)
      callBack.onFailure(ApolloNetworkException(message, e))
```

and

```
          val operationName = request.operation.name().name()
          val message = "Failed to execute http call for operation '$operationName'"
          logger.e(e, message)
          callBack.onFailure(ApolloNetworkException(message, e))
```

in the ApolloServerInterceptor Kotlin class.

**Comment by martinbonnin** [comment]
👍  I'm in the middle of something so can't easily switch branched right now but will rebase a bit later. 

Thanks for the contribution!

**Comment by semaphore3000** [comment]
Thank you, too.

**Pull request merged**